lib/kubernetes-deploy/kubernetes_resource/stateful_set.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+require 'kubernetes-deploy/kubernetes_resource/base_set'
+module KubernetesDeploy
+  class StatefulSet < BaseSet
+    TIMEOUT = 5.minutes
+
+    def sync
+      raw_json, _err, st = kubectl.run("get", type, @name, "--output=json")
+      @found = st.success?
+
+      if @found
+        stateful_data = JSON.parse(raw_json)
+        @current_generation = stateful_data["metadata"]["generation"]
+        @observed_generation = stateful_data["status"]["observedGeneration"]
+        @desired_replicas = stateful_data["spec"]["replicas"].to_i
+        @rollout_data = stateful_data["status"].slice("replicas")
```

Comment:
This is slicing out the exact same thing you're using for `desired_replicas`, so if I'm not mistaken, your current success condition will always pass as soon as the first new replica is created. I see in [the docs](https://v1-7.docs.kubernetes.io/docs/api-reference/v1.7/#statefulsetstatus-v1beta1-apps) that SS have a distinctive set of fields, two of which reference `Revision` (that's probably referring to [controllerRevision](https://github.com/kubernetes/kubernetes/pull/45867)?). Please look into and document which combinations of these fields we should look at to most reliably detect rollout completion.